docs/advanced/checkpointing.md

```diff
@@ -0,0 +1,108 @@
+# Checkpointing
+
+To perform stateful operations reliably and with consistent outputs, Quix Streams applications need to periodically save their state stores to disk and commit processed offsets to Kafka.
+
+We call this process a “checkpointing”.
+
+The goal of checkpointing is to ensure that applications can recover after failures and reprocess records from Kafka producing the same results as if the failure never happened.
```

Comment:
Not arguing the main goal here, but increasing processing speed is also a pretty large intention of this.

This might not immediately concern the average user, but might be worth mentioning if someone is reading this doc to understand the architecture more and wonder why we only commit every so often.

If you did want to mention it, the "how" part could probably be put elsewhere to keep this section concise.

Reply:
It's a valid point, but in this exact case I'd rather focus on one thing - reliability - and how it's been achieved.